### PR TITLE
added ability to inject custom drivers

### PR DESCRIPTION
**Pull request by maximgubar**
| Q | A
| --- | ---
| Branch? | master 
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Tests pass? | yes
| Fixed tickets | 
| License | MIT
| Doc PR | 

this PR makes possible to declare and inject custom drivers for the bundle.
validation on strict match for 'gd', 'imagick', 'gmagick' has been removed

### CODE REVIEW

File: Binary/Locator/FileSystemInsecureLocator.php [hunk 1]
```diff
@@ -21,9 +21,9 @@ class FileSystemInsecureLocator extends FileSystemLocator
      */
     protected function generateAbsolutePath(string $root, string $path): ?string
     {
-        if (false === mb_strpos($path, '..'.DIRECTORY_SEPARATOR) &&
-            false === mb_strpos($path, DIRECTORY_SEPARATOR.'..') &&
-            false !== file_exists($absolute = $root.DIRECTORY_SEPARATOR.$path)
+        if (false === mb_strpos($path, '..'.\DIRECTORY_SEPARATOR) &&
```
Comment (lsmith77) [inline Binary/Locator/FileSystemInsecureLocator.php]: why the backslashes here?

Comment (lsmith77) [inline Binary/Locator/FileSystemInsecureLocator.php]: why the backslashes here?

File: Binary/Locator/FileSystemLocator.php [hunk 2]
```diff
@@ -60,7 +60,7 @@ public function locate(string $path): string
      */
     protected function generateAbsolutePath(string $root, string $path): ?string
     {
-        if (false !== $absolute = realpath($root.DIRECTORY_SEPARATOR.$path)) {
+        if (false !== $absolute = realpath($root.\DIRECTORY_SEPARATOR.$path)) {
```
Comment (lsmith77) [inline Binary/Locator/FileSystemLocator.php]: why the backslashes here?

File: DependencyInjection/Factory/Loader/FileSystemLoaderFactory.php [hunk 3]
```diff
@@ -127,7 +127,7 @@ private function getBundleResourcePaths(ContainerBuilder $container)
         }
 
         return array_map(function ($path) {
-            return $path.DIRECTORY_SEPARATOR.'Resources'.DIRECTORY_SEPARATOR.'public';
+            return $path.\DIRECTORY_SEPARATOR.'Resources'.\DIRECTORY_SEPARATOR.'public';
```
Comment (lsmith77) [inline DependencyInjection/Factory/Loader/FileSystemLoaderFactory.php]: why the backslashes here?

File: Tests/AbstractTest.php [hunk 4]
```diff
@@ -50,8 +50,8 @@
 
     protected function setUp()
     {
-        $this->fixturesPath = realpath(__DIR__.DIRECTORY_SEPARATOR.'Fixtures');
-        $this->temporaryPath = sys_get_temp_dir().DIRECTORY_SEPARATOR.'liip_imagine_test';
+        $this->fixturesPath = realpath(__DIR__.\DIRECTORY_SEPARATOR.'Fixtures');
```
Comment (lsmith77) [inline Tests/AbstractTest.php]: why the backslashes here?

**Comment by lsmith77** [comment]
maybe we should have a 2.1 branch for this? or should we actually start using master again for future minor/major new versions?

File: Binary/Locator/FileSystemInsecureLocator.php (hunk 1 above)
Comment (maximgubar) [reply Binary/Locator/FileSystemInsecureLocator.php]: it is global constant and php-cs doesn't like it without backslash

Comment (maximgubar) [reply Binary/Locator/FileSystemInsecureLocator.php]: was fixed in [1107](https://github.com/liip/LiipImagineBundle/pull/1107).
will remove backslashes.

**Comment by maximgubar** [comment]
@lsmith77 we started to use `master` branch again, the base branch of this PR was changed.

**Pull request merged**